src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs

```diff
@@ -48,9 +48,11 @@ public class VsPackageInstaller : IVsPackageInstaller2
             _sourceRepositoryProvider = sourceRepositoryProvider;
             _settings = settings;
             _solutionManager = solutionManager;
-            _projectContext = new VSAPIProjectContext();
             _packageServices = packageServices;
             _deleteOnRestartManager = deleteOnRestartManager;
+
+            _projectContext = new Lazy<INuGetProjectContext>(() => new VSAPIProjectContext());
```

Comment:
This looks like it would work, but should the ServiceLocator call in VSAPIProjectContext be lazy instead? It might be more beneficial to do that instead since it looks like it would rarely be used on this logger.